### ISSUE TITLE
create Admin controller

### ISSUE THREAD

**Issue by HellHunterMax** [Code2Gether-Discord/JokesOnYou#58]
Admin should be able to create roles and assign roles to users.

admin should be able to use the Users controller.

**Comment by HellHunterMax** [comment]
I think we need a Roles Controller, not admin ?

**Comment by chrisK00** [comment]
> I think we need a Roles Controller, not admin ?

It depends. Isnt admin the only one that can create roles? Not sure what purpose rolescontroller would have since the auth will take care of authentication

**Comment by HellHunterMax** [comment]
> > I think we need a Roles Controller, not admin ?
> 
> It depends. Isnt admin the only one that can create roles? Not sure what purpose rolescontroller would have since the auth will take care of authentication

Well admin should be a role by itself and admin should be authorized to do the creation and modification of roles ?
Ill have to read up on roles.
Do you still have the link for the roles

**Comment by HellHunterMax** [comment]
scrap that we should have fixed roles ?

**Comment by chrisK00** [comment]
> > > I think we need a Roles Controller, not admin ?
> > 
> > 
> > It depends. Isnt admin the only one that can create roles? Not sure what purpose rolescontroller would have since the auth will take care of authentication
> 
> Well admin should be a role by itself and admin should be authorized to do the creation and modification of roles ?
> Ill have to read up on roles.
> Do you still have the link for the roles

Yes so on the AdminController we use Authorize and the "Admin" so only a authorized admin can reach the admin methods rn only add role since we will have mods aswell so a ModeratorController too. Authorizing is just done by adding that annotation in a nutshell. You might be overthinking it
